Stratis.Bitcoin/RPC/Controllers/FullNodeController.cs

```diff
@@ -1,51 +1,132 @@
 ﻿using Microsoft.AspNetCore.Mvc;
-using NBitcoin;
+using Microsoft.Extensions.PlatformAbstractions;
+using Stratis.Bitcoin.Configuration;
 using Stratis.Bitcoin.RPC.Models;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Stratis.Bitcoin.BlockStore;
+using Stratis.Bitcoin.MemoryPool;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Utilities;
 
 namespace Stratis.Bitcoin.RPC.Controllers
 {
     //TODO: Need to be extensible, should be FullNodeController
-    public partial class ConsensusController : Controller
+    public partial class FullNodeController : Controller
     {
+        private Builder.IFullNode _FullNode;
+        private NodeSettings _Settings;
+        private NBitcoin.Network _Network;
+        private ConsensusValidator _ConsensusValidator;
+        private NBitcoin.ChainBase _Chain;
+        private ChainBehavior.ChainState _ChainState;
+        private BlockStoreManager _BlockManager;
+        private MempoolManager _MempoolManager;
+        private Connection.ConnectionManager _ConnectionManager;
+
+        public FullNodeController(
```

Comment:
I expect some of these constructor parameters to be required instead of nullable and thus this constructor to have a Guard method for those.